### PR TITLE
[16_5] Add documentation and support for archlinux to `bin/format`

### PR DESCRIPTION
**Pull request by Oyyko**
## Why you open this Pull Request?

Add documentation for bin/format. And add support for archlinux.

## What work have you done in the current Pull Request?

- [x] Add doc for developers to use `bin/format`.
- [x] Add support for archlinux.

### CODE REVIEW

File: docs/guide/Develop_on_Linux.md [hunk 1]
```diff
@@ -43,6 +43,10 @@ xmake install research
 xmake run research
 ```
 
+
+### Preparation work before PR
```
Comment (darcy-shen) [inline docs/guide/Develop_on_Linux.md:47]: duplicated doc, it should be maintained in a separated file, like how to run unit tests

**Comment by Oyyko** [comment]
Now fixed. Do you think it is OK?

File: docs/zh/guide/Contribution.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# 贡献
+## PR前的准备工作
+安装elvish并执行`bin/format`脚本来格式化代码。如果您想手动执行`clang-format`,您可以自行安装`clang-format`并手动格式化相关代码。
```
Comment (darcy-shen) [inline docs/zh/guide/Contribution.md:3]: elvish -> Elvish

半角的逗号换成全角的逗号

**Review by darcy-shen — APPROVED** [review]
LGTM

File: docs/guide/Contribution.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Contribution
+## Preparation work before PR
+Install elvish and execute the `bin/format` script to format the code. If you want to execute `clang-format` manually, you can install `clang-format` yourself and format the relevant code manually.
```
Comment (darcy-shen) [inline docs/guide/Contribution.md:3]: elvish -> Elvish

**Pull request merged**